scenario-discovery (#4)
josephguillaume/uncertainfutures: _theory/CART.md

---
layout: concept
title: CART
permalink: /theory/CART/

> excerpt: CART (Classification and Regression Trees) [1] is a binary recursive partitioning procedure and machine learning methods for building prediction models from data.

Reviewer: Can we give a more plain language summary that hints at how it is used for scenario discovery?

Author: Would this suggestion be clearer:
CART is a binary and iterative procedure and machine learning method used in scenario discovery to identify vulnerable scenarios for policy measures.

purpose: >
  <p>CART is a non-parametric, rule-based machine learning method, that can be used for <a href="../theory/scenario-discovery/">scenario discovery</a> , to inductively discover relationships between input predictor variables and a target attribute.</p>

>   <p>It presents scenarios as decision trees </p>.

Reviewer: Needs to be expanded on a little

>   <p>It presents scenarios as decision trees </p>.

Reviewer: ```suggestion
  <p>It presents scenarios as decision trees.</p>
```

requirements: >
  <p>Two main requirements [2]:<ul>
  <li>Generation of datasets of uncertain input parameters with binary outputs (e.g., acceptable/unacceptable) based on simulation models </li>

>   <li>Definition of splitting criteria to identify the output class associated to a combination of inputs (i.e., minimise impurities)</li></ul></p>

Reviewer: These two requirements are written fairly technically. Is there a way to simplify them?

>   <p> Does not require model assumptions [14]</p> 

Reviewer: What is meant by this?

Reviewer: ```suggestion
  # TODO: Revisit later: <p> Does not require model assumptions [14]</p> 
```

>   <p> Can model complex relationships automatically [14]</p>

Reviewer: What type of complex relationships are being referred to here?

decomposition: >
  <p>Three major steps could be identified:<ol>

>   <li>A 'splitting' stage in which a database has successively its input space, one parameter split at a time, into two subsets. The partitioning is based on minimising impurities on each part of the split and makes use of Gini index for impurity measure and to partition attributes. The process is run iteratively, leading to the growth of the decision tree until describing entirely pure regions. [2] </li>

Reviewer: Word missing here.
The meaning of "pure regions" is not clear.

>   <li>A 'pruning' stage in which some nodes of the final tree are recombined with assigning each terminal node to one class representing the most appropriate target value and ultimately, to find the tree with the best predictive power [2]. Examples of pruning methods include cost-complexity pruning, weakest link pruning [8]</li>

Reviewer: ```suggestion
  <li>A 'pruning' stage in which some nodes of the final tree are recombined, assigning each terminal node to one class representing the most appropriate target value, aiming to find the tree with the best predictive power [2]. Examples of pruning methods include cost-complexity pruning and weakest link pruning [8]</li>
```

>   <li>User selection stage among decision trees of the number of groups (i.e., scenarios) to investigate, based on the desired balance between predictive performance and interpretability</li>

Reviewer: ```suggestion
  <li>User selection stage among decision trees of the number of groups (i.e., scenarios) to investigate, based on the desired balance between predictive performance and interpretability</li>
  # TODO: A figure could be added in future
```

  </ol></p>

>   <p> An alternative form of CART for modelling multi-level data with binary outcomes (M-CART) combines multi-logistic regression (M-logit) to CART using an expectation maximisation algorithm [14].</p>

Reviewer: ```suggestion
  <p> An alternative form of CART for modelling multi-level data with binary outcomes (M-CART) combines multi-logistic regression (M-logit) with CART using an expectation maximisation algorithm [14].</p>
```

tools:
  - slug: rpart

>     name: Recursive Partitioning (rpart) and Regression Trees package

Reviewer: ```suggestion
    name: Recursive Partitioning (rpart) and Regression Trees R package
```

    description: Recursive partitioning for classification, regression and survival trees [3]
  - slug: openMORDM
    name: openMORDM

>     description: Software developed as an R package for generating or loading datasets [4]

Reviewer: ```suggestion
    description: An R package for <a href="../MORDM">many-objective robust decision making</a> [4]
```

  - slug: Orange
    name: Orange

>     description: open source data mining package that consist of a C++library with python bidings [5][6]

Reviewer: ```suggestion
    description: open source data mining package that consists of a C++ library with python bindings [5,6]
```

>   - slug: MathWorks

Reviewer: ```suggestion
  - slug: MATLAB
```

>     name: MathWorks

Reviewer: ```suggestion
    name: MATLAB
```

>     description: Matlab and Statistics and Machine Learning Toolbox including CART functions for scenario vulnerability identification [7]

Reviewer: ```suggestion
    description: Matlab Statistics and Machine Learning Toolbox includes CART functions [7]
```

also_see:
  - permalink: /tools/PRIM/
    name: PRIM
    description: A common alternative algorithm for scenario discovery
  - permalink: /theory/exploratory-modelling/

>     name: exploratory modelling

Reviewer: ```suggestion
    name: Exploratory modelling
```

  - >
    [7] MathWorks Inc. (2015). MATLAB and Statistics and Machine Learning Toolbox.
  - >
    [8] Hastie, T., Tibshirani, R., & Friedman, J. (2009). The elements of statistical learning: Data mining, inference, and prediction. Springer Science & Business Media.
  - >
    [9] Almeida, S., Holcombe, E. A., Pianosi, F., & Wagener, T. (2017). Dealing with deep uncertainties in landslide modelling for disaster risk reduction under climate change. Natural Hazards and Earth System Sciences, 17(2), 225–241.
  - >
    [10] Erechtchoukova, M. G., Khaiter, P. A., & Saffarpour, S. (2016). <a href="https://doi.org/10.1007/s11269-016-1423-6">Short-Term Predictions of Hydrological Events on an Urbanized Watershed Using Supervised Classification. </a>Water Resources Management, 30(12), 4329–4343.
  - >
    [11] García, V. J., Márquez, C. O., Isenhart, T. M., Rodríguez, M., Crespo, S. D., & Cifuentes, A. G. (2019).<a href="https://doi.org/10.1016/j.heliyon.2019.e02701"> Evaluating the conservation state of the páramo ecosystem: An object-based image analysis and CART algorithm approach for central Ecuador. </a> Heliyon, 5(10), e02701.
  - >

>     [12] Gerst, M. D., Wang, P., Roventini, A., & Dosi, G. (2012). <a href="http://www.iemss.org/society/index.php/iemss-2012-proceedings">Integrated assessment of mitigation strategies using an agent-based model of the linked energy, economic, and climate system.</a> International Environmental Modelling Software Society (IEMSs), International Congress on Environmental Modelling and Software. Managing Resources of a Limited Planet, Sixth Biennial Meeting, Leipzig, Germany.

Reviewer: This link is broken

Author: Remove the link from the reference

  - >
    [13] Kumar, R., Jaiswal, A., & Rai, D. (2013). Privacy Preserving CART Algorithm over Vertically Partitioned Data. International Journal of Innovative Research in Computer and Communication Engineering, 1(9).
  - >
    [14] Lin, S. (2018). <a href="http://hdl.handle.net/1969.1/173483">A New Multilevel Cart Algorithm and Its Application in Propensity Score Analysis</a> [Texas A & M University].
  - >
    [15] Naghibi, S. A., Pourghasemi, H. R., & Dixon, B. (2015). <a href="https://doi.org/10.1007/s10661-015-5049-6">GIS-based groundwater potential mapping using boosted regression tree, classification and regression tree, and random forest machine learning models in Iran.</a> Environmental Monitoring and Assessment, 188(1), 44.
  - >
    [16] Hadka, D., Herman, J., Reed, P., & Keller, K. (2015).<a href="https://doi.org/10.1016/j.envsoft.2015.07.014"> An open source framework for many-objective robust decision making.</a> Environmental Modelling & Software, 74, 114–129.
examples:

>   - Landslide hazards prediction in the Caribbean under geotechnical, hydrological and geometrical parameters uncertainty [9]

Reviewer: Not water related? Delete?
```suggestion
  - Landslide hazards prediction in the Caribbean under geotechnical, hydrological and geometrical parameters uncertainty [9]
```

>   - Short-term prediction of flood events in a highly urbanised watershed [10]

Reviewer: ```suggestion
  - CART provides short-term prediction that a flood will occur in a highly urbanised watershed, identifying the conditions leading to that prediction [10]
```
Correct? It would be good to be clear on the role of CART within the example, and emphasise any link scenario discovery

>   - Evaluation of the impacts of land cover/land use change (including waterbodies) and climate change on the conservation of ecosystem services [11]

Reviewer: What is the role of CART and link to scenario discovery in this example?

>   - Application for scenario discovery coupled with ABM for climate mitigation strategies [12]

Reviewer: ```suggestion
  - Application for scenario discovery coupled with an agent-based model for climate mitigation strategies [12]
```

Reviewer: Need a bit more detail about how this it is used for scenario discovery

>   - Application for data mining and privacy preserving [13]

Reviewer: Not related to water. Delete?
```suggestion
```

>   - Groundwater mapping based on GIS integration of CART [15]

Reviewer: This sounds like a pure prediction application of CART, not related to scenario discovery - I would prefer to delete?

>   - Application to the "lake problem" to identify robust water pollution management policies [16]

Reviewer: ```suggestion
  - Application to the <a href="../../examples/lake-problem/">"lake problem"</a> to identify robust water pollution management policies [16]
```

Reviewer: Would be good to indicate how it is used to identify robust policies - are the variables being split policies?
